packages/dev/core/src/Engines/Processors/Expressions/shaderDefineExpression.ts

```diff
@@ -34,6 +37,17 @@ export class ShaderDefineExpression {
     }
 
     public static infixToPostfix(infix: string): string[] {
+        // Is infix already in cache
+        const existedResult = this._InfixToPostfixCache.get(infix);
+        if (existedResult) {
+            return existedResult;
+        }
+
+        // Is infix contain any operator
+        if (!infix.includes(")") && !infix.includes("(") && !infix.includes("||") && !infix.includes("&&")) {

```

Comment:
Does this one really make a difference? There will be 4 string searches each time we execute this line...

If there's no operator in the string, the main loop below should be pretty fast as we avoid executing almost all the code.

Reply:
> Does this one really make a difference? There will be 4 string searches each time we execute this line...

This check cut 23128 calls and pass 182 for next computation.
I think, the main performance boost related to anonyumous functions that creates right after this check, I talk about this part:
![image](https://github.com/BabylonJS/Babylon.js/assets/2697890/724fec3b-eed9-4b72-b1e4-882b46f0fcc9)

But let's drop my thoughts and give a true check with measurement tool.
Results with `if (!infix.includes(')') && !infix.includes('(') && !infix.includes('||') && !infix.includes('&&'))` :
Average computation time for all calls while the scene is loading: 19.4801ms
Without this condition I got the next result: 26.2751ms for all calls

Than let's see the stats of every calls before this check:

- infix contains `(`: 2 times
- infix contains `)`: 2 times
- infix contains `||`: 40 times
- infix contains `&&`: 138 times
- infix does not contain any operators: 14902 times

Based on this data I tried to simplify the condition to `!infix.includes('||') && !infix.includes('&&')` and got 17.9525ms. Looks like the brackets is not popular operators, so I'll push this condition as update.
Than I switched `&&` with `||` in the condition and got 15.8724ms for all calls.

P.S. Each measurement is average based on 30 runs.

Reply:
Actually, you can't skip testing "(" and ")", else you may return an infix string with parenthesis that have not been processed.

Maybe declaring the `pushOperand`, `push`, `peek` and `pop` functions as true functions of the class would be a better win?

Reply:
I returned brackets. And trying to make anonymous functions as a true function does not change performance in my case, but it make the code much harder because they are too sideeffectes.

Reply:
Is there another comments to this PR? Should I change something other to make PR better?